pkg/debug/checkups/connectivity.go

```diff
@@ -0,0 +1,119 @@
+package checkups
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/kolide/launcher/pkg/agent/types"
+)
+
+const requestTimeout = time.Second * 5
+
+type Connectivity struct {
+	k       types.Knapsack
+	status  Status
+	summary string
+}
+
+func (c *Connectivity) Name() string {
+	return "Check communication with Kolide"
+}
+
+func (c *Connectivity) Run(ctx context.Context, fullFH io.Writer) error {
+	//if !c.k.KolideHosted() {
+	//	c.status = Unknown
+	//	c.summary = "not kolide hosted"
+	//	return nil
+	//}
+
+	httpClient := &http.Client{Timeout: requestTimeout}
+
+	hosts := map[string]string{
+		"device server":  c.k.KolideServerURL(),
+		"control server": c.k.ControlServerURL(),
+	}
+
+	for n, v := range hosts {
+		fmt.Fprintf(fullFH, "Response from %s / %s:\n", n, v)
+		if err := checkKolideServer(c.k, httpClient, fullFH, v); err != nil {
+			fmt.Fprintf(fullFH, "\n")
+			return fmt.Errorf("%s(%s): %w", n, v, err)
+		}
+		fmt.Fprintf(fullFH, "\n")
+	}
+
+	c.status = Passing
+	c.summary = "successfully connected to device and control server"
+
+	return nil
+}
+
+func (c *Connectivity) ExtraFileName() string {
+	return "responses.txt"
+}
+
+func (c *Connectivity) Status() Status {
+	return c.status
+}
+
+func (c *Connectivity) Summary() string {
+	return c.summary
+}
+
+func (c *Connectivity) Data() any {
+	return nil
```

Comment:
Why is this nil?

Reply:
Because it's not plumbed to logs yet, and I'm not sure how I want to structure it